backend/function-apps/dataflows/import/migrate-consolidations.ts

```diff
@@ -1,45 +1,159 @@
+  }
+}
+
+/**
+ * queuePages
+ *
+ * @param params
+ * @param invocationContext
+ * @returns
+ */
+async function queuePages(predicate: AcmsPredicate, invocationContext: InvocationContext) {
+  const logger = ApplicationContextCreator.getLogger(invocationContext);
+  const appContext = await ApplicationContextCreator.getApplicationContext({
+    invocationContext,
+    logger,
+  });
+  const controller = new AcmsOrdersController();
+
+  try {
+    const leadCaseIds = await controller.getLeadCaseIds(appContext, predicate);
+
+    if (!leadCaseIds.length) {
+      logger.debug(
+        MODULE_NAME,
+        `No lead case Ids to queue for consolidation migration for division ${predicate.divisionCode} chapter ${predicate.chapter}`,
+      );
+      return;
+    }
 
-  df.app.activity(QUEUE_MIGRATION_ACTIVITY, {
-    extraOutputs: [migrationQueue],
-    handler: queueMigrateConsolidation,
+    logger.debug(MODULE_NAME, `Putting ${leadCaseIds.length} consolidations in the queue.`);
+
+    // Transform the lead case IDs into the queue items.
+    const queueItems = [];
+    for (let i = 0; i < leadCaseIds.length; i++) {
+      const leadCaseIdString = leadCaseIds[i].toString();
+      const queueItem: AcmsEtlQueueItem = {
+        divisionCode: predicate.divisionCode,
+        chapter: predicate.chapter,
+        leadCaseId: leadCaseIdString,
+      };
+      queueItems.push(queueItem);
+    }
+
+    // Slice the queue items into pages for the PAGE queue.
+    const pages = [];
+    for (let i = 0; i < queueItems.length; i += PAGE_SIZE) {
+      pages.push(queueItems.slice(i, i + PAGE_SIZE));
+    }
+    invocationContext.extraOutputs.set(PAGE, pages);
+  } catch (originalError) {
+    const error = getCamsError(originalError, MODULE_NAME, 'Failed to get lead case ids.');
+    logger.camsError(error);
+  }
+}
+
+async function migrateConsolidation(item: AcmsEtlQueueItem, context: InvocationContext) {
+  const logger = ApplicationContextCreator.getLogger(context);
+  const appContext = await ApplicationContextCreator.getApplicationContext({
+    invocationContext: context,
+    logger,
   });
-  df.app.activity(FLATTEN_BOUNDING_ARRAYS_ACTIVITY, flattenBoundingArrays);
+  try {
+    const controller = new AcmsOrdersController();
+
+    const result = await controller.migrateConsolidation(appContext, item.leadCaseId);
+    logger.debug(MODULE_NAME, `Migrate consolidation of ${item.leadCaseId}: ${result.success}.`);
 
-  app.storageQueue(MIGRATE_CONSOLIDATION, {
-    connection: 'AzureWebJobsStorage',
-    queueName: migrationQueue.queueName,
-    extraOutputs: [successQueue, failQueue],
-    handler: migrateConsolidation,
+    if (!result.success && result.error) {
+      throw result.error;
+    }
+  } catch (originalError) {
+    const errorMessage = {
+      message: item,
+      error: getCamsError(originalError, MODULE_NAME),
+    };
+    logger.error(MODULE_NAME, JSON.stringify(item), errorMessage.error);
```

Comment:
This is putting a stringified item in the message argument of the log. Do we want to do that? Instead we could do something like:
```suggestion
    logger.error(MODULE_NAME, originalError.name, { item, errorMessage.error, error: errorMessage.error });
```

Not sure about what the message should be, so consider `originalError.name` to be a placeholder.

Reply:
Removed the JSON serialization. The data for the item will go into the HARD_STOP queue to examine.